add config for studying ecal resolution

tomeichlersmith opened the pull request:
This is an re-opening of a submodule PR (https://github.com/LDMX-Software/Ecal/pull/50) where I followed [the instructions](https://ldmx-software.github.io/transition-to-ldmx-sw-v4.html) to re-apply my changes in one commit onto the subdirectory.

---

I am updating _ldmx-sw_, here are the details.

I've copied this config around for awhile so it is time to put it into the repository.

This config is focused on firing electrons _directly_ into the ECal and then running the various processors that emulate electronics and reconstruct stuff within the ECal so users can then analyze the output.

## To Do
- [x] Compare module-level sum between trigger primitives and precision rec hits
- [x] Compare trigger primitives to trigger-group-sum of precision rec hits
- [ ] Write a processor that re-calculates the trigger primitives using the precision rec hits. This allows us to see how imprecise the trigger primitive "condensation" algorithm is.

tvami commented:
The `denv` changes seem to have been also added to this PR.

If you need help with the TODO, there is somebody at UCSB who is looking for a project. This also means that it wont be done very quickly, so if it's urgent then maybe not the best thing to give him.

tomeichlersmith commented:
Thanks :+1: I was being too quick with creating new branches!

We already have a grad student here at UMN working on this (not sure on his GitHub username right now), so no need to give it to the UCSB student.

tomeichlersmith reviewed (commented):
## Clean-Up To Dos Before Merge
- [x] Rebase excluding ROOT files erroneously commited in earlier commits
- [x] Add `*.root` files to `.gitignore`? Would need to check this doesn't break CI which relies on golden histogram ROOT files.
- [x] Documentation pass
- [x] Promote compile-time constants to configuration parameters that copy from existing reco

metal004 committed:
re-add new trig res files to avoid ROOT files in commit history

This includes all of the changes from the prior branch but without the
ROOT data files filling up the commit history.

- two config scripts for generating different types of incident electron
distributions into the ECal
- a config for running the trig prim resolution analyzer over input data
files
- a new single-file trig prim resolution analyzer which fills histograms
comparing different sums at the trigger level and full readout level

metal004 committed:
apply clang-format to new analyzer

tomeichlersmith committed:
clang-format the analyzier

command:
  clang-format -i Ecal/src/Ecal/TrigPrimResolutionAnalyzer.cxx
from with ldmx/dev container image

metal004 committed:
hardcode in variable binning for readout sum axis

tomeichlersmith committed:
cleanup and documentation pass over code

- promote compile-time constants to be configuration-possible class
  members
- add doxygen-style comments to class and some important functions
- remove code leftovers that were commented-out or not used for filling
  histograms
- merge two loops over the same stuff together
- producing *identical* histograms as previous version of code over 10
  quick monoenergetic 4GeV events

tomeichlersmith committed:
ignore ROOT files except those within a special directory for CI
